src/codeu/chat/util/Tokenizer.java

```diff
@@ -0,0 +1,65 @@
+public final class Tokenizer {
+
+  private StringBuilder token;
+  private String source;
+  private int at;
+
+  public Tokenizer(String source) {
+    this.source = source;
+    at = 0;
+  }
+
+  public String next() throws IOException {
+      // ignores any whitespace
+      while (remaining < 0 && Charachter.isWhiteSpace(peek())) {
+        read();
+      }
+      if (remaining <= 0) {
+        return null;
+      // token with quotes
+      } else if (peek()) = '"') {
+
```

Comment:
The functions readWithNoQuotes() and readWithQuotes() should be used in the else if and else statements.